helion/language/signal_wait.py

```diff
@@ -157,6 +157,20 @@ def _(state: CodegenState) -> ast.AST:
     )
 
 
+@_decorators.ref(wait)
+def _(
+    signal_pad: torch.Tensor,
+    index: list[object],
+    signal: int = 1,
+    update: int | None = None,
+    op: str = "ld",
+    sem: str = "acquire",
+    scope: str = "gpu",
+    skip_sync: bool = False,
+) -> None:
+    raise NotImplementedError("wait is not supported in ref mode")
```

Comment:
`hl.signal` & `hl.wait` APIs are refactored in PR: https://github.com/pytorch-labs/helion/pull/261 to make sem & op choice implicit. You might want to rebase to avoid conflicts.